docker/dockerfile.merlin

```diff
@@ -89,7 +89,7 @@ RUN pip install cupy-cuda117 nvidia-pyindex pybind11 pytest protobuf transformer
 RUN pip install betterproto cachetools graphviz nvtx scipy sklearn
 RUN pip install tritonclient[all] grpcio-channelz fiddle wandb
 RUN pip install git+https://github.com/rapidsai/asvdb.git@main
-RUN pip install xgboost lightgbm treelite==2.3.0 treelite_runtime==2.3.0
+RUN pip install xgboost==1.6.2 lightgbm treelite==2.3.0 treelite_runtime==2.3.0
```

Comment:
could be worth adding a comment next to this to explain why we're pinning xgboost/treelite (both required to match  a version of the Triton FIL backend